app/controllers/ems_common.rb

```diff
@@ -766,4 +801,29 @@ def show_list_link(ems, options = {})
                                     :action     => "show_list",
                                     :id         => ems.id))
   end
+
+  #
+  # Returns the path of the template that generates the provider specific part of the new and edit
+  # provider dialogs.
+  #
+  # @param type [String] The type of provider, for example `rhevm`.
+  # @return [String] The path of the template.
+  #
+  def properties_template(type)
+    template = "/providers/#{type}/properties"
+    unless lookup_context.exists?(template, [], true)
+      template = '/providers/common/properties'
+    end
+    template
+  end
+
+  #
+  # Generates the URL path that the client should use to request the provider specific part of the
+  # new and edit provider dialogs.
+  #
+  # @return [String] The absolute path that the client should use.
+  #
+  def properties_path
+    '/ems_infra/properties'
```

Comment:
I see - the way we are defining routes is not using resource based routing in which does not allow us to use named routes... I'll add it to as a new issue